### PR TITLE
Fix benchmark + add it to `npm test`

### PR DESCRIPTION
**Pull request by bajtos**
- add `benchmark` package to `npm test` 
- add autocannon typings 
- improve the code handling worker startup to detect & reporterrors 
- make "create todo" scenario pass

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: benchmark/src/worker.ts [hunk 1]
```diff
@@ -18,13 +22,29 @@ async function main() {
...
   app.bind('datasources.config.db').to({connector: 'memory'});
 
+  // overwrite the error logger to print all failed requests, including 4xx
+  app.bind(RestBindings.SequenceActions.LOG_ERROR).to(logAllErrors);
+
   await app.start();
 
   // Tell the master thread what is our URL
-  console.log(app.restServer.url);
+  console.log('Server listening at', app.restServer.url);
```
Comment (jannyHou) [inline benchmark/src/worker.ts]: nitpick: add a space after `at`?

**Comment by agnes512** [comment]
Does the CI failure have something to do with the changes?

File: benchmark/src/worker.ts (hunk 1 above)
Comment (bajtos) [reply benchmark/src/worker.ts]: Please note `console.log` is already adding one space character to delimit arguments.

**Comment by bajtos** [comment]
> Does the CI failure have something to do with the changes?

My changes are adding benchmark tests to the test suite and these newly added tests are failing on come of our CI platforms. I am going to increase the test timeout to see if that solves the issue.

File: benchmark/src/worker.ts (hunk 1 above)
Comment (bajtos) [reply benchmark/src/worker.ts]:
```
$ node
Welcome to Node.js v12.13.1.
Type ".help" for more information.
> console.log('Server listening at', 'http://localhost')
Server listening at http://localhost
undefined
```

**Pull request merged**